.github/workflows/phpunit.yml

```diff
@@ -0,0 +1,64 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+name: "PHPUnit tests"
+
+on:
+    pull_request:
+    push:
+        branches:
+            - "master"
+
+jobs:
+    phpunit:
+        name: "PHPUnit tests"
+
+        runs-on: ${{ matrix.operating-system }}
+
+        strategy:
+            matrix:
+                dependencies:
+                    - "lowest"
+                    - "highest"
+                    - "locked"
+                php-version:
+                    - "7.2"
+                    - "7.3"
+                    - "7.4"
+                operating-system:
+                    - "ubuntu-latest"
+                    - "windows-latest"
+
+        steps:
+            - name: "Checkout"
+              uses: "actions/checkout@v2"
+
+            - name: "Install PHP"
+              uses: "shivammathur/setup-php@v2"
+              with:
+                  coverage: "pcov"
+                  php-version: "${{ matrix.php-version }}"
+                  ini-values: memory_limit=-1
+
+            - name: "Cache dependencies"
+              uses: "actions/cache@v2"
+              with:
+                  path: |
+                      ~/.composer/cache
```

Comment:
Does Composer on Windows store the caches here?

Reply:
Probably not, but given that we use a `composer.lock`, perhaps we can avoid this caching step overall?

Reply:
Well, there are "lowest" and "highest" deps in the matrix, which trigger `composer update`, which uses cache...

Yet [there's a canonical way to handle this](https://github.com/shivammathur/setup-php#cache-composer-dependencies).